flyteidl/protos/flyteidl/plugins/kubeflow/mpi.proto

```diff
@@ -40,4 +40,7 @@ message DistributedMPITrainingReplicaSpec {
 
   // MPI sometimes requires different command set for different replica groups
   repeated string command = 5;
+
+  // Node selectors for the replica group
+  map<string, string> node_selectors = 6;
```

Comment:
I think the change should, unfortunately, be more involved than this... In general we try to abstract away k8s (and infrastructure) concepts when users write code.
For instance, when you write an `@Task` you can add `@task(interruptible=True)` and not have to think about the right node selectors to add... the administrator would configure those at propeller level... This makes the code more portable and separates infra concerns.

The way I would implement this change is:
1. Add 
    ```
        oneof is_interruptible {
            bool interruptible = 6;
       }
    ````
    This allows you to know if the field is set and if the value it was set to was False... That or create an enum with 3 states (UNSET, TRUE, FALSE)
2. When the pod is being built here: https://github.com/flyteorg/flyte/blob/master/flyteplugins/go/tasks/plugins/k8s/kfoperators/common/common_operator.go#L270, you can see it checks for the interruptible field from task execution metadata: https://github.com/flyteorg/flyte/blob/55a67f23f5d8e44bc7d70f738b0fafe954e0bf94/flyteplugins/go/tasks/pluginmachinery/flytek8s/pod_helper.go#L222... so what you want to do is refactor some of these common functions to allow you to call them from here: replicaSpec, err := common.ToReplicaSpec(ctx, taskCtx, kubeflowv1.MPIJobDefaultContainerName) based on the above flag (whether that's set at the worker pod vs launcher pod)

This way the user can just say: `@task(config=TFTraining(worker=PodSpec(interruptible=True)))` or something and it would use the same settings configured for `@task(interruptible=True)` just for the worker...

Happy to jump on a call if you want to discuss further...

cc @Future-Outlier